tests/UnitTests.vala

```diff
@@ -3,15 +3,247 @@ namespace Vamp {
+                    assert_cmpstr (package.name, GLib.CompareOperator.EQ, "test-project");
+                    assert_cmpstr (package.version, GLib.CompareOperator.EQ, "0.0.1");
+                    assert_cmpstr (package.description, GLib.CompareOperator.EQ, "A Test Project");
+                } catch (FileError e) {
+                    error (e.message);
+                }
             });
 
-            Test.add_func ("/vamp/bar", () => {
-                assert_true (true);
+            Test.add_func ("/vamp/deserialize_full_config", () => {
+                string package_contents;
+                try {
+                    bool did_open = FileUtils.get_contents (TestConfig.FULL_TEST_PACKAGE_FILE, out package_contents);
+                    if (!did_open) {
+                        error ("Failed to get contents from: %s".printf (TestConfig.FULL_TEST_PACKAGE_FILE));
+                    }
+
+                    Vamp.Package package = desrialise_package_config (package_contents);
+                    assert_cmpstr (package.name, GLib.CompareOperator.EQ, "test-project");
+                    assert_cmpstr (package.version, GLib.CompareOperator.EQ, "0.0.1");
+                    assert_cmpstr (package.description, GLib.CompareOperator.EQ, "A Test Project");
+                    assert_cmpstrv (package.keywords.to_array (), {"test", "project", "fake", "mock"});
+                    assert_cmpstr (package.homepage, GLib.CompareOperator.EQ, "https://wwww.test.com");
+
+                    // We assert objects this way so that:
+                    // 1. We know the exact value of each object property
+                    // 2. We can easily update our asserts as we update the
+                    // object's properties.
+                    assert_full_config_bugs (package.bugs);
+                    assert_cmpstr (package.license, GLib.CompareOperator.EQ, "MIT");
+                    assert_full_config_author (package.author);
+                    assert_full_config_contributors (package.contributors);
+                    assert_full_config_funding (package.funding);
+                    assert_cmpstrv (package.files.to_array (), { "./main-module/**/*", "./extra-module/**/*"});
+                    assert_full_config_respository (package.repository);
+                    assert_full_config_dependencies (package.dependencies);
+                    assert_full_config_dev_dependencies (package.dev_dependencies);
+                    assert_full_config_optional_dependencies (package.optional_dependencies);
+                } catch (FileError e) {
+                    error (e.message);
+                }
+            });
+
+            Test.add_func ("/vamp/serialize_basic_config", () => {
+                Vamp.Package package = new Vamp.Package ();
+                package.name = "basic-project";
+                package.version = "1.0.0";
+                package.description = "A basic project";
```

Comment:
Prefer to use the syntax of object initializers, so that you avoid repeating `package` over and over again.

```suggestion
                Vamp.Package package = new Vamp.Package () {
                    name = "basic-project",
                    version = "1.0.0",
                    description = "A basic project"
                }
```